featurebyte/api/observation_table.py

```diff
@@ -393,6 +433,10 @@ def upload(
             Purpose of the observation table.
         primary_entities: Optional[List[str]]
             List of primary entities for the observation table.
+        target_column: Optional[str]
+            Name of the column in the observation table that stores the target values.
+            The target column name must match an existing target namespace in the catalog.
+            The primary entities must match the entities in the target namespace.
```

Comment:
Target namespace also tracks a dtype optionally. Should that be validated as well?

Reply:
Indeed. Will add validation of dtype